File: src/com/betterzhang/learnkotlin/java/Book.java
package com.betterzhang.learnkotlin.java;

/**
 * Created by IntelliJ IDEA.
 * Author : Andrew Zhang
 * Email  : betterzhang.dev@gmail.com
 * Time   : 2017/06/28 上午 9:54
 * Desc   : description
 */
public class Book implements Readable {

    @Override
    public String getContent() {
        return "Hello Java";
    }

}

File: src/com/betterzhang/learnkotlin/kotlin/Book.kt
package com.betterzhang.learnkotlin.kotlin

/**
 * Created by IntelliJ IDEA.
 * Author : Andrew Zhang
 * Email  : betterzhang.dev@gmail.com
 * Time   : 2017/06/28 上午 9:56
 * Desc   : description
 */
class Book : Readable {

//    override fun getContent(): String {
//        return "Hello Kotlin"
//    }

    override fun getContent(): String = "Hello Kotlin"

    override fun getVersion(): Int {
        return super.getVersion()
    }

}